src/receive_imf.rs

```diff
@@ -778,6 +778,11 @@ async fn add_parts(
         }
     }
 
+    if chat_id.is_none() && is_mdn {
```

Comment:
E.g. Gmail saves outgoing MDNs to the sentbox, so this should be moved out of `if mime_parser.incoming {`. This allows to remove `is_mdn` checks for the common code below.